pkg/operator/controllers/dnsmasq/machineconfigpool_controller.go

```diff
@@ -30,35 +31,29 @@ type MachineConfigPoolReconciler struct {
 }
 
 func NewMachineConfigPoolReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface) *MachineConfigPoolReconciler {
-	return &MachineConfigPoolReconciler{
+	r := &MachineConfigPoolReconciler{
 		AROController: base.AROController{
-			Log:    log,
-			Client: client,
-			Name:   MachineConfigPoolControllerName,
+			Log:         log.WithField("controller", machineConfigPoolControllerName),
+			Client:      client,
+			Name:        machineConfigPoolControllerName,
+			EnabledFlag: controllerEnabled,
 		},
 		dh: dh,
 	}
+	r.Reconciler = r
+	return r
 }
 
 // Reconcile watches MachineConfigPool objects, and if any changes,
 // reconciles the associated ARO DNS MachineConfig object
-func (r *MachineConfigPoolReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	instance, err := r.GetCluster(ctx)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-
-	if !instance.Spec.OperatorFlags.GetSimpleBoolean(operator.DnsmasqEnabled) {
-		r.Log.Debug("controller is disabled")
-		return reconcile.Result{}, nil
-	}
+func (r *MachineConfigPoolReconciler) ReconcileEnabled(ctx context.Context, request ctrl.Request, instance *arov1alpha1.Cluster) (ctrl.Result, error) {
+	var err error
```

Comment:
nit: where possible, don't allow short-lived values to escape their smallest possible scope. For instance, something like

```go
if err := something(); err != nil {
}
```

Can help to reduce mistakes - one global `err` that gets mutated over and over again won't be caught by the compiler if, for instance, one of the individual mutations fails to check `err != nil` once more.